### PR TITLE
Jonalex

### PR DESCRIPTION
**Pull request by Jonathan-Alexander**

### CODE REVIEW

**Comment by Jonathan-Alexander** [comment]
Adds libraries for Linear/Logistic Regression Analysis, Correlation Analysis, and Jupyter notebook for q2 (still cleaning up graphs)

File: rotten_tomatoes/correlation_analysis.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import seaborn as sns
+import pandas as pd
+import numpy as np
+
+class CorrelationAnalysis:
+    
+    def __init__(self, d):
```
Comment (kathrynline) [inline rotten_tomatoes/correlation_analysis.py]: You might want to add some types here. Something like 
`def __init__(self, d: pd.DataFrame) -> None:`

And everything is going to need docstrings.

File: rotten_tomatoes/correlation_analysis.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        self.data = d
+        
+    def corr_matrix(self, subset = None):
```
Comment (kathrynline) [inline rotten_tomatoes/correlation_analysis.py]: Would add an assert that the columns of subset are in data.

File: rotten_tomatoes/correlation_analysis.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        if subset == None:
```
Comment (kathrynline) [inline rotten_tomatoes/correlation_analysis.py]: Nit: You should only have to use one "if" statement here. 
```
if subset: 
    return self.data[subset].corr()
   
return self.data.corr()
```

File: rotten_tomatoes/correlation_analysis.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            data = self.data
+        else:
+            data = self.data[subset]
+            
+        return data.corr()
+    
+    def plot_heatmap(self, subset = None):
+        sns.heatmap(self.corr_matrix(subset).round(2), annot=True, vmax=1, vmin=-1)
+        return
```
Comment (kathrynline) [inline rotten_tomatoes/correlation_analysis.py]: Unnecessary return; None return is implicit in python

File: rotten_tomatoes/correlation_analysis.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    
+    def corr_coef(self, a, b):
+        return data[a].corr(data[b])
```
Comment (kathrynline) [inline rotten_tomatoes/correlation_analysis.py]: Would add asserts that a and b are columns in data

File: rotten_tomatoes/regression_analysis.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import pandas as pd
+import numpy as np
+import math
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.metrics import mean_squared_error, mean_absolute_error
+import matplotlib.pyplot as plt
+
+class RegressionAnalysis:
+    def __init__(self, X, y, is_categorical, random_state = 123, test_size=0.25):
```
Comment (kathrynline) [inline rotten_tomatoes/regression_analysis.py]: Need docstrings

File: rotten_tomatoes/regression_analysis.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            X_test = X_test.reshape(-1, 1)
+
+        return X_train, X_test, y_train, y_test
+    
+    def set_X_cols(self, cols):
+        '''
+        set a subset of columns to be used as input
+        '''
+        self.col_indexes = [self.X.columns.get_loc(c) for c in cols]
+        return
```
Comment (kathrynline) [inline rotten_tomatoes/regression_analysis.py]: Don't need explicit return

File: rotten_tomatoes/regression_analysis.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    def fit_train(self):
+        '''
+        Fits the model (Linear or Logistic)
+        using the supplied columns, or all the columns in X if not given
+        
+        @param id_cols - list of columns in X to use when fitting
+        @return void
+        '''
+        self.model_ = self.model_.fit(self.X_train(), self.y_train_)
+        return
```
Comment (kathrynline) [inline rotten_tomatoes/regression_analysis.py]: Don't need explicit return

File: rotten_tomatoes/correlation_analysis.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import seaborn as sns
```
Comment (kathrynline) [inline rotten_tomatoes/correlation_analysis.py]: Can you go ahead and stick all these classes in utils?

File: rotten_tomatoes/correlation_analysis.py (hunk 2 above)
Comment (Jonathan-Alexander) [reply rotten_tomatoes/correlation_analysis.py]: If the columns aren't a subset you hit a pandas assert failure which felt more transparent to the user. Do you think an additional assert is necessary?

File: rotten_tomatoes/correlation_analysis.py (hunk 1 above)
Comment (Jonathan-Alexander) [reply rotten_tomatoes/correlation_analysis.py]: Updated to include type and docstring

File: rotten_tomatoes/correlation_analysis.py (hunk 3 above)
Comment (Jonathan-Alexander) [reply rotten_tomatoes/correlation_analysis.py]: Updated

File: rotten_tomatoes/correlation_analysis.py (hunk 4 above)
Comment (Jonathan-Alexander) [reply rotten_tomatoes/correlation_analysis.py]: Updated

File: rotten_tomatoes/correlation_analysis.py (hunk 2 above)
Comment (kathrynline) [reply rotten_tomatoes/correlation_analysis.py]: Ok sure, that's fair.

**Review by kathrynline — APPROVED** [review]

**Pull request merged**